src/tasks/utils.ts

```diff
@@ -0,0 +1,106 @@
+import type { TaskId, TaskIdEncoded, TaskPriority } from './types';
+import type { NodeId } from '../nodes/types';
+import { IdInternal, IdSortable } from '@matrixai/id';
+import lexi from 'lexicographic-integer';
+
+/**
+ * Generates TaskId
+ * TaskIds are lexicographically sortable 128 bit IDs
+ * They are strictly monotonic and unique with respect to the `nodeId`
+ * When the `NodeId` changes, make sure to regenerate this generator
+ */
+function createTaskIdGenerator(lastTaskId?: TaskId) {
+  const generator = new IdSortable<TaskId>({
+    lastId: lastTaskId,
+  });
+  return () => generator.get();
+}
```

Comment:
Do we care if the `NodeId` has changed? If we only seed the generator from the last `TaskId` then it will never update the `NodeId`. Would that be a problem?

Reply:
No I removed it purposely. It made it easier to see what the task ids are. This is only required to have node id, if the intention is to compare/track tasks across nodes. And this is not relevant to the PK tasks system.